treeherder/model/migrations/0022_modify_bugscache_and_bugjobmap.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.9 on 2018-01-31 21:41
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('model', '0021_remove_revision_hash'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='bugscache',
+            name='whiteboard',
+            field=models.CharField(blank=True, default='', max_length=100),
+        ),
+        migrations.AlterField(
+            model_name='bugjobmap',
+            name='bug_id',
+            field=models.PositiveIntegerField(db_column='bug_id'),
```

Comment:
Thank you for using that guide to preserving data!

Django migrations can occasionally be buggy for the MySQL backend (more people use the Postgres backend as I understand it), so we tend to always check the output of `sqlmigrate` when making migrations to ensure it's doing the right thing.

Doing that now, I got:

```
vagrant ~/treeherder $ ./manage.py sqlmigrate model 0022
BEGIN;
--
-- Add field whiteboard to bugscache
--
ALTER TABLE `bugscache` ADD COLUMN `whiteboard` varchar(100) DEFAULT  NOT NULL;
ALTER TABLE `bugscache` ALTER COLUMN `whiteboard` DROP DEFAULT;
--
-- Alter field bug_id on bugjobmap
--
DROP INDEX `bug_job_map_bug_id_013db14b` ON `bug_job_map`;
--
-- Rename field bug_id on bugjobmap to bug
--
--
-- Alter unique_together for bugjobmap (1 constraint(s))
--
--
-- Alter field bug on bugjobmap
--
CREATE INDEX `bug_job_map_bug_id_013db14b` ON `bug_job_map` (`bug_id`);
ALTER TABLE `bug_job_map` ADD CONSTRAINT `bug_job_map_bug_id_013db14b_fk_bugscache_id` FOREIGN KEY (`bug_id`) REFERENCES `bugscache` (`id`);
--
-- Alter field time on push
--
CREATE INDEX `push_time_eb18a3b0` ON `push` (`time`);
COMMIT;
```

In it, the index on `bug_id` is being dropped and then recreated again. I believe this is because the field definition on this intermediate step is missing the `db_index=True`. 

If I change that line to this:
```python
models.PositiveIntegerField(db_index=True, db_column='bug_id'),
```

...then the generated migration looks fine (aside from the weird whiteboard `DEFAULT NOT NULL` followed immediately by `DROP DEFAULT` thing, but that must be a separate Django bug and not anything we need to worry about from a performance of applying the migration in production point of view):

```
BEGIN;
--
-- Add field whiteboard to bugscache
--
ALTER TABLE `bugscache` ADD COLUMN `whiteboard` varchar(100) DEFAULT  NOT NULL;
ALTER TABLE `bugscache` ALTER COLUMN `whiteboard` DROP DEFAULT;
--
-- Alter field bug_id on bugjobmap
--
--
-- Rename field bug_id on bugjobmap to bug
--
--
-- Alter unique_together for bugjobmap (1 constraint(s))
--
--
-- Alter field bug on bugjobmap
--
ALTER TABLE `bug_job_map` ADD CONSTRAINT `bug_job_map_bug_id_013db14b_fk_bugscache_id` FOREIGN KEY (`bug_id`) REFERENCES `bugscache` (`id`);
--
-- Alter field time on push
--
CREATE INDEX `push_time_eb18a3b0` ON `push` (`time`);
COMMIT;
```